src/json_utils.hpp

```diff
@@ -64,9 +64,14 @@ std::optional<T> extractTypedValue(nlohmann::json::const_iterator it) {
         if (!it->is_boolean())
             return std::nullopt;
         return it->template get_ref<const nlohmann::json::boolean_t&>();
+    } else {
+        // This should never be reached for supported types
+        static_assert(std::is_same_v<T, std::string> || std::is_same_v<T, int> ||
+                          std::is_same_v<T, int64_t> || std::is_same_v<T, double> ||
+                          std::is_same_v<T, bool>,
+                      "Unsupported type for extractTypedValue");
```

Comment:
minor: my understanding is that this should never be reached and is equivalent to:
```suggestion
        static_assert(false, "Unsupported type for extractTypedValue");
```
...which communicates this assumption slightly better. Otherwise, one needs to review all branches before reaching out this conclusion.